store/src/store.rs

```diff
@@ -130,6 +128,8 @@ pub trait ChainStore: Sync + Send {
     fn is_uncle(&self, hash: &H256) -> bool;
     // Get cellbase by block hash
     fn get_cellbase(&self, hash: &H256) -> Option<Transaction>;
+    // Get the ancestor of a base block by a block number
+    fn get_ancestor(&self, base: &H256, number: BlockNumber) -> Option<Header>;
```

Comment:
Is this a refactoring or a rebase conflict that we move this method from chain provider to store?

Reply:
This is a refactoring, we need this method in DaoCalculator but due to dependency problem, DaoCalculator can only get hold of store, not ChainProvider, I checked the implementation of get_ancestor and noticed it only needs a store, hence I moved the method.